llvm/tools/llvm-reduce/deltas/ReduceDIMetadata.cpp

```diff
@@ -65,12 +65,14 @@ void identifyUninterestingMDNodes(Oracle &O, MDNodeList &MDs) {
     SmallVector<Metadata *, 16> TN;
     for (size_t I = 0; I < Tup->getNumOperands(); ++I) {
       // Ignore any operands that are not DebugInfo metadata nodes.
-      if (isa_and_nonnull<DINode>(Tup->getOperand(I)))
-        // Don't add uninteresting operands to the tuple.
-        if (!O.shouldKeep())
-          continue;
-
-      TN.push_back(Tup->getOperand(I));
+      if (Metadata *Op = Tup->getOperand(I).get()) {
+        if (isa<DINode>(Op) || isa<DIGlobalVariableExpression>(Op)) {
```

Comment:
I believe it is possible to find nulls in the MD tree. Given the brute-force approach of this pass, I think `isa_and_nonnull` makes more sense.

Reply:
I've guarded it with `if (Metadata *Op = Tup->getOperand(I).get())` above this line

Reply:
Ah yes. That's fine, then. One nit: The braces on this if statement are not necessary.